diff --git a/multimod/src/db/src/main/java/ch/mycompany/test/db/App.java b/multimod/src/db/src/main/java/ch/mycompany/test/db/App.java
index 57b70a8..d6e8952 100755
--- a/multimod/src/db/src/main/java/ch/mycompany/test/db/App.java
+++ b/multimod/src/db/src/main/java/ch/mycompany/test/db/App.java
@@ -12 +12 @@ public class App
-        double d = Double.longBitsToDouble(i);  // Noncompliant
+        double g = Double.longBitsToDouble(i);  // Noncompliant
@@ -16 +16 @@ public class App
-        System.out.println( "Hello World! " + d);
+        System.out.println( "Hello World! " + g);
@@ -17,0 +18,10 @@ public class App
+
+	public void foo () {
+		int f = 42;
+
+        try {
+        } catch (Throwable t) {
+        }
+	}
+
+	private void foo3() { }
diff --git a/multimod/src/db/src/test/java/ch/mycompany/test/db/AppTest.java b/multimod/src/db/src/test/java/ch/mycompany/test/db/AppTest.java
deleted file mode 100755
index 4545ecc..0000000
--- a/multimod/src/db/src/test/java/ch/mycompany/test/db/AppTest.java
+++ /dev/null
@@ -1,40 +0,0 @@
-package ch.mycompany.test.db;
-
-import junit.framework.Test;
-import junit.framework.TestCase;
-import junit.framework.TestSuite;
-
-
-/**
- * Unit test for simple App.
- */
-public class AppTest
-    extends TestCase
-{
-    /**
-     * Create the test case
-     *
-     * @param testName name of the test case
-     */
-    public AppTest( String testName )
-    {
-        super( testName );
-    }
-
-    /**
-     * @return the suite of tests being tested
-     */
-    public static Test suite()
-    {
-        return new TestSuite( AppTest.class );
-    }
-
-    /**
-     * Rigourous Test :-)
-     */
-    public void testApp()
-    {
-        ch.mycompany.test.db.App.main(new String[]{});
-        assertTrue( true );
-    }
-}
diff --git a/multimod/src/gui/src/main/java/ch/mycompany/test/gui/App.java b/multimod/src/gui/src/main/java/ch/mycompany/test/gui/App.java
index 356f4b7..6ca26ea 100755
--- a/multimod/src/gui/src/main/java/ch/mycompany/test/gui/App.java
+++ b/multimod/src/gui/src/main/java/ch/mycompany/test/gui/App.java
@@ -8,0 +9 @@ public class App
+        double d = Double.longBitsToDouble(42);  // Noncompliant
@@ -26,2 +27,2 @@ public class App
-        int i = 100023;
-        System.exit(-1);
+        int i = 100024;
+         System.exit(-1);
@@ -40,0 +42,7 @@ public class App
+
+    private void foo2() {
+        int baba = 5555;
+        //TODO fix this
+    }
+
+
diff --git a/multimod/src/gui/src/main/java/ch/mycompany/test/gui/StashTag.java b/multimod/src/gui/src/main/java/ch/mycompany/test/gui/StashTag.java
new file mode 100755
index 0000000..3052a0b
--- /dev/null
+++ b/multimod/src/gui/src/main/java/ch/mycompany/test/gui/StashTag.java
@@ -0,0 +1,26 @@
+package ch.mycompany.test.gui;
+
+public class StashTag {
+    private final String id;
+    private final String displayId;
+
+    public StashTag(String id, String displayId) {
+   	//FIXME
+        id = id;
+        this.id = id;
+        this.displayId = displayId;
+    }
+
+    // value is being used by Velocity template
+    @SuppressWarnings("UnusedDeclaration")
+    public String getId() {
+        return id;
+    }
+
+    // value is being used by Velocity template
+    @SuppressWarnings("UnusedDeclaration")
+    public String getDisplayId() {
+	String s = null;
+        return displayId;
+    }
+}
diff --git a/sonar.json b/sonar.json
index 69165e1..b8fc582 100644
--- a/sonar.json
+++ b/sonar.json
@@ -0,0 +1 @@
+<<<<<<< destination:d657f52f8f651d212c3399785bff035ae29bdd99
@@ -3,0 +5,5 @@
+=======
+{
+	'sonarHost': 'https://myhost.com',
+	'sonarProjectKey': 'ch.mycompany.test:multimod:develop'
+>>>>>>> source:c778919ef079ad35688c7f5dc2ecb42aee57ae56
